locallib.php

```diff
@@ -340,8 +340,11 @@ function tool_securityquestions_inject_navigation_node($navigation, $user, $user
  * Forces redirect to the set_responses page if users havent answered enough questions
  */
 function require_question_responses() {
-    global $USER;
-    global $DB;
+    global $USER, $DB, $SESSION;
+
+    if (property_exists($SESSION, 'presentedresponse') && !get_config('tool_securityquestions', 'mandatory_questions')) {
```

Comment:
This is fine but seems like this is more consistent with elsewhere?

 !empty($SESSION->presentedresponse)

also if I'm going to use a bunch of config instead of 

get_config('tool_securityquestions', 'mandatory_questions')

I'll do

$config = get_config('tool_securityquestions')

and then later $config->mandatory_questions  it's just less boilerplate so a little nicer to read